### PR TITLE
feat: repo-local colima

### PR DESCRIPTION
**Pull request by joshuarli**
so... people are starting to run into some problems with colima 0.6.2 and we should update to latest anyways (some CVEs are fixed)

i've moved this to the new repo-local approach (see https://github.com/getsentry/devenv/pull/78) which is better and more flexible

running all newest devservices on macos 14.3 works on my M1

after this, planning on releasing a new version along with adding to sentry devenv/config.ini

### CODE REVIEW

**Review by IanWoodard — APPROVED** [review]
Once we get the tests passing, looks good to go

**Comment by joshuarli** [comment]
after some discussion with @asottile-sentry i'm removing the sync and bootstrap unittests as they don't really do much beyond calling functions, and unittests are useful for individual functions

i'll follow up soon with a unittest for run_procs but want to get this out today

File: devenv/lib/config.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+from __future__ import annotations
```
Comment (joshuarli) [inline devenv/lib/config.py:1]: unittest coverage for this will be addressed in https://github.com/getsentry/devenv/pull/80

**Review by asottile-sentry — APPROVED** [review]

**Pull request merged**